### PR TITLE
scram: fix client final message

### PR DESCRIPTION
**Pull request by dotnwat**
The client should send r=ServerNonce rather than r=Salt

### CODE REVIEW

File: src/kpro_scram.erl [hunk 1]
```diff
@@ -67,7 +67,7 @@ parse(#{ sha := Sha
   <<MyNonce:?MY_NONCE_LEN/binary, _/binary>> = ServerNonce,
   Salt = base64:decode(Salt0),
   SaltedPassword = hi(Sha, Password, Salt, Iterations),
-  FinalNoProof = bin([?CHANNEL_BINDING, ",r=", Salt]),
+  FinalNoProof = bin([?CHANNEL_BINDING, ",r=", ServerNonce]),
```
Comment (dotnwat) [inline src/kpro_scram.erl:70]: here is an example from the widely used franz-go kafka client:

https://github.com/twmb/franz-go/blob/master/pkg/sasl/scram/scram.go#L221-L223

**Comment by dotnwat** [comment]
bump @zmstone

**Comment by dbegunkov** [comment]
Thank you for your fix. It's work!

**Comment by astorre88** [comment]
Hey, guys! Please merge it ASAP)

**Comment by zmstone** [comment]
sorry for the late response.
thank you for the fix.

**Comment by zmstone** [comment]
published 4.1.3

**Pull request merged**